PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -1453,6 +1484,173 @@ struct LambdaTracksExtProducer {
+
+    // Fill pair density
+    histos.fill(HIST("RecoCorr/h2f_n2_phiphi_") + HIST(SubDirHist[part_pair]), cent, RecoDecay::constrainAngle(RecoDecay::phi(pr1)), RecoDecay::phi(pr2));
+    histos.fill(HIST("RecoCorr/h2f_n2_dphi_") + HIST(SubDirHist[part_pair]), cent, RecoDecay::constrainAngle(RecoDecay::phi(pr1) - RecoDecay::phi(pr2), -PI));
+  }
+
+  template <ParticleType part, typename T>
+  void analyzeSingles(T const& tracks)
+  {
+    static constexpr std::string_view SubDirHist[] = {"LaP", "LaM"};
+    int ntrk = 0;
+    for (auto const& track : tracks) {
+      // Count tracks
+      ++ntrk;
+
+      // Get four-momentum of lambda
+      std::array<float, 4> l = {MassLambda0, track.px(), track.py(), track.pz()};
+      std::array<float, 4> p = {MassProton, track.prPx(), track.prPy(), track.prPz()};
+      std::array<float, 3> v;
+      getBoostVector(l, v, cInvBoostFlag);
+      boost(p, v);
+
+      // Fill single histograms
+      histos.fill(HIST("RecoCorr/h2f_n1_phi_") + HIST(SubDirHist[part]), cent, RecoDecay::constrainAngle(RecoDecay::phi(p)));
+    }
+  }
+
+  template <ParticlePairType partpair, bool samelambda, typename T>
+  void analyzePairs(T const& trks_1, T const& trks_2)
+  {
+    for (auto const& trk_1 : trks_1) {
+      for (auto const& trk_2 : trks_2) {
+        // check for same index for Lambda-Lambda / AntiLambda-AntiLambda
+        if (samelambda && ((trk_1.index() == trk_2.index()))) {
+          continue;
+        }
+        fillPairHistos<partpair>(trk_1, trk_2);
+      }
+    }
+  }
+
+  // Initialize tables
+  using LambdaCollisions = aod::LambdaCollisions;
+  using LambdaTracks = soa::Join<aod::LambdaTracks, aod::LambdaTracksExt>;
+
+  SliceCache cache;
+  Partition<LambdaTracks> partLambdaTracks = (aod::lambdatrack::v0Type == (int8_t)kLambda) && (aod::lambdatrackext::trueLambdaFlag == true) && (aod::lambdatrack::v0PrmScd == (int8_t)kPrimary);
+  Partition<LambdaTracks> partAntiLambdaTracks = (aod::lambdatrack::v0Type == (int8_t)kAntiLambda) && (aod::lambdatrackext::trueLambdaFlag == true) && (aod::lambdatrack::v0PrmScd == (int8_t)kPrimary);
+
+  void processDataReco(LambdaCollisions::iterator const& collision, LambdaTracks const&)
+  {
+    // assign centrality
+    cent = collision.cent();
+
+    auto lambdaTracks = partLambdaTracks->sliceByCached(aod::lambdatrack::lambdaCollisionId, collision.globalIndex(), cache);
+    auto antiLambdaTracks = partAntiLambdaTracks->sliceByCached(aod::lambdatrack::lambdaCollisionId, collision.globalIndex(), cache);
+
+    analyzeSingles<kLambda>(lambdaTracks);
+    analyzeSingles<kAntiLambda>(antiLambdaTracks);
+
+    if (cAnaPairs) {
+      analyzePairs<kLambdaAntiLambda, false>(lambdaTracks, antiLambdaTracks);
+      analyzePairs<kLambdaLambda, true>(lambdaTracks, lambdaTracks);
+      analyzePairs<kAntiLambdaAntiLambda, true>(antiLambdaTracks, antiLambdaTracks);
+    }
+  }
+
+  PROCESS_SWITCH(LambdaSpinCorrelation, processDataReco, "Process for Data and MCReco", true);
```

Comment:
Be aware that if you don't want to run this task and the flag is configured to `false` the workflow will hang
If you don't want this to happen declare additionally an empty `process` function